cyberdrop_dl/clients/scraper_client.py

```diff
@@ -76,47 +78,112 @@ async def on_request_end(*args):
+            response: str = solution.get("response")
+            response_url = solution.get("url") if with_response_url else None
+            if response_url:
+                response_url = URL(response_url)
+            return response, response_url
+
+    @limiter
+    async def get_soup(
+        self,
+        domain: str,
+        url: URL,
+        client_session: aiohttp_client_cache.CachedSession,
+        origin: ScrapeItem | URL | None = None,
+        with_response_url: bool = False,
+        cache_disabled: bool = False,
+    ) -> BeautifulSoup:
+        """Returns a BeautifulSoup object from the given URL."""
+        if cache_disabled:
+            async with client_session.disabled():
+                async with client_session.get(
                     url,
-                    client_session,
-                    origin,
-                )
-                # retry request with flaresolverr cookies
-                if self.client_manager.check_ddos_guard(soup) or self.client_manager.check_cloudflare(soup):
-                    if not retry:
-                        raise DDOSGuardError(message="Unable to access website with flaresolverr cookies") from None
-                    return await self.get_soup(domain, url, client_session, origin, with_response_url, retry=False)
+                    headers=self._headers,
+                    ssl=self.client_manager.ssl_context,
+                    proxy=self.client_manager.proxy,
+                ) as response:
+                    try:
+                        await self.client_manager.check_http_status(response, origin=origin)
+                    except DDOSGuardError:
+                        await self.client_manager.manager.cache_manager.request_cache.delete_url(url)
+                        response, response_URL = await self.flaresolverr(
+                            domain,
+                            url,
+                            origin=origin,
+                            with_response_url=with_response_url,
+                        )
+                        if with_response_url:
+                            return BeautifulSoup(response, "html.parser"), response_URL
+                        return BeautifulSoup(response, "html.parser")
+
+                    content_type = response.headers.get("Content-Type")
+                    assert content_type is not None
+                    if not any(s in content_type.lower() for s in ("html", "text")):
+                        raise InvalidContentTypeError(
+                            message=f"Received {content_type}, was expecting text", origin=origin
+                        )
+                    text = await CachedStreamReader(await response.read()).read()
+                    if with_response_url:
+                        return BeautifulSoup(text, "html.parser"), URL(response.url)
+                    return BeautifulSoup(text, "html.parser")
+        else:
+            async with client_session.get(
```

Comment:
Instead of using 2 `if` branches with basically the same code, use a custom context manager that replaces `disabled` and accepts a `bool` to disable cache conditionally:

```python
@asynccontextmanager
async def cache_control_manager(self, disabled: bool = False, client_session: ClientSession):
    client_session.cache.disabled = disabled
    yield
    client_session.cache.disabled = False
'''

Reply:
Use for all `scrape_client` functions